EasyPost/Calculation/Rates.cs

```diff
@@ -17,148 +15,29 @@ public static class Rates
-                        continue;
-                    }
-
-                    if (excludeServices.Contains(rate.Service!.ToLowerInvariant()))
-                    {
-                        // If the rate's service is in the list of services to exclude, skip it
-                        continue;
-                    }
-                }
-
-                if (lowestRate == null)
-                {
-                    // if lowest rate is null, set it to this rate
-                    lowestRate = rate;
-                    continue;
-                }
-
-                if (rate.Price == null || lowestRate.Price == null)
-                {
-                    // if the rate or lowest rate doesn't have a price, throw an exception
-                    throw new FilteringError(Constants.ErrorMessages.NullObjectComparison);
-                }
-
-                float rateValue = float.Parse(rate.Price, NumberStyles.Any, CultureInfo.InvariantCulture);
-                float lowestRateValue = float.Parse(lowestRate.Price, NumberStyles.Any, CultureInfo.InvariantCulture);
-
-                if (rateValue >= lowestRateValue)
-                {
-                    // if this rate is greater than or equal to the lowest rate, skip it
-                    continue;
-                }
-
-                // if we made it this far, this rate is the lowest rate, set it to the lowest rate
-                lowestRate = rate;
-            }
-
-            if (lowestRate == null)
-            {
-                // if we didn't find a rate, throw an exception
-                throw new FilteringError(string.Format(CultureInfo.InvariantCulture, Constants.ErrorMessages.NoObjectFound, "rates"));
-            }
-
-            // return the lowest rate
-            return lowestRate;
-        }
+        /// <summary>
+        ///     Get the lowest rate from a list of ephemeral rates.
+        /// </summary>
+        /// <param name="rates">List of rates to parse.</param>
+        /// <param name="includeCarriers">Carriers to include in the filter.</param>
+        /// <param name="includeServices">Services to include in the filter.</param>
+        /// <param name="excludeCarriers">Carriers to exclude in the filter.</param>
+        /// <param name="excludeServices">Services to exclude in the filter.</param>
+        /// <returns>Lowest rate matching the filter.</returns>
+        [Obsolete("This method is deprecated. Please use EasyPost.Utilities.Rates.GetLowestStatelessRate() instead. This method will be removed in a future version.", false)]
+        public static EasyPost.Models.API.Beta.StatelessRate GetLowestStatelessRate(IEnumerable<EasyPost.Models.API.Beta.StatelessRate> rates, List<string>? includeCarriers = null, List<string>? includeServices = null, List<string>? excludeCarriers = null, List<string>? excludeServices = null) => Utilities.Rates.GetLowestStatelessRate(rates, includeCarriers, includeServices, excludeCarriers, excludeServices);
```

Comment:
Don't deprecate a function that never existed please, let's just remove it outright. No need to add dead code that is about to get yanked out.